### DISCUSSION TITLE
Queue Mangement: Clear Queue

### DISCUSSION THREAD

**Discussion by hrast01** [jcorporation/myMPD#558] [Q&A] [answered]
I'm really confused about Clear Queue?  I don't understand the logic behind hitting Clear Queue, and it leaving a non-playing song in queue (even if its been partially played and is now paused).  I would think Clear would remove everything, regardless of current state of the player.  The weirdness is only re-enforced when a second Clear Queue will remove the track left behind.  Is there a use case I'm not seeing?

**Comment by jcorporation** [comment, marked as answer]
The button itself should be intelligent:
- clear if no song is playing
- clear if only one song is in queue 
- crop if a song is playing

Clear with no conditions can be done with the clear button in the dropdown of this intelligent button.
